### PR TITLE
added decorator

### PR DESCRIPTION
**Pull request by Mithi-code**
### OOP school library: decorate a class.

- Implement composition (as an example of the design pattern).
- Run a program using the command line.
- Prepare CapitalizeDecorator and TrimmerDecorator.
- Turn Person class to Nameable.

### CODE REVIEW

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Corrector
+  def correct_name(name)
+    name.capitalize.slice(0, 10)
+  end
+end
```
Comment (EricMbouwe) [inline corrector.rb]: - [ ] Here you should have an Interface (a Namable class) with a method 'correct_name' **rasing** an error. Let's fix that

File: person.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require_relative('./corrector')
 class Person
   attr_accessor :name, :age
```
Comment (EricMbouwe) [inline person.rb]: - [ ] The `Person` class should inherit from the `Nameable` class

File: person.rb [hunk 3]
```diff
@@ -16,5 +18,9 @@ def of_age?
   def can_use_services?
     @parent_permision || of_age?
   end
+
+  def validate_name
+    @name = @corrector.correct_name(@name)
+  end
   private :of_age?
```
Comment (EricMbouwe) [inline person.rb:27]: - [ ] You should create a `Decorator` class that inherits from Namable and implement the `correct_name ` method that returns the result of the `correct_name ` method of the @namable.
- [ ] You should create a `CapitalizeDecorator`  class and `TrimmerDecorator` class that inherits from `Decorator`

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @Mithi-code,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight!
- Good PR ✔️
- Good commits ✔️

# Required Changes ♻️ ⛔️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
Hi **@[Mithi-code](https://github.com/Mithi-code),** 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
appr

# Status: Approved ✔️✔️✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ] It will be better to use the given name if any. Here you were requested to name your class `Decorator` and you are using `Corrector`. This will avoid confusion and show that you pay attention to details.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**